### ISSUE TITLE
clarify how syntactic shortcut for options are used with non-string option values

### ISSUE THREAD

**Issue by rdeltour** [xproc/1.0-specification#161]
Section [4.8.1 Syntactic shortcut for options](http://www.w3.org/TR/xproc20/#option-shortcut) seems clear enough for atomic types, at least `xs:string`, but it's unclear how to define more exoticly-typed options, like maps.

See also #146.

**Comment by rdeltour** [comment]
For a map, am I correct assuming you'd have to use an AVT in this case?

``` xml
<ex:stepType map-option="{map {
  "Su" : "Sunday",
  "Mo" : "Monday",
  "Tu" : "Tuesday",
  "We" : "Wednesday",
  "Th" : "Thursday",
  "Fr" : "Friday",
  "Sa" : "Saturday"
}}"/>
```

But for "castable" option value, e.g. `xs:integers`, is the following possible?

``` xml
<ex:stepType int-option="1"/>
```

Or should it be interpreted as "a non-string value can only be set with AVT"?

**Comment by ndw** [comment]
You're right, some clarity is required. I think there are two possibilities.
1. The value of a shortcut attribute is an XPath expression
2. The value is a string in which AVTs can occur

I have a slight preference for the former, but in the interest of backwards compatibility (and compatibility with attribute values on other elements), I think we should probably do the latter.

**Comment by rdeltour** [comment]
Backwards compatibility is a compelling argument, so let's assume the value is a string. The next thing to clarify is whether any kind of type coercion is performed by the processor. For this latter, again two possibilities:
-  only `xs:string` options can be set that way
- type coercion is performed to set options of type `xs:string`, `xs:boolean`, `xs:double`, `xs:integer`, ???

**Comment by ndw** [comment]
Proposal:
1. Curly braces in AVTs atomize and produce strings.
2. A shortcut option that begins with a single ^ treats the remaining characters in the string as a macro

`<ex:step opt="3">`

Sets opt = string value 3.

`<ex:step opt="^3">`

Sets opt = integer value 3 (or decimal or whatever XPath does with the expression 3)

`<ex:step opt='^map{"one": 1, "two": 2}'>`

Sets opt = a map with two keys.

Yes, we're inventing a new escaping mechanism.

P.S. If you want a literal ^, you have to escape it `opt="^'^'"`

**Comment by ndw** [comment]
See also #80 for the questions of coercion.
